R/highlight.R

```diff
@@ -121,6 +121,23 @@ line_col <- function(x) {
   data.frame(line, col)
 }
 
+# utils::getParseData will truncate very long strings or tokens;
+# this function checks for that and uses the slow
+# utils::getParseText function when necessary.
+
+getFullParseData <- function(x) {
+  res <- utils::getParseData(x)
+
+  truncated <- res$terminal &
+    substr(res$text, 1, 1) == "[" &
+    nchar(res$text) > 1
+
+  if (any(truncated))
+    res[truncated, "text"] <- utils::getParseText(res, res$id[truncated])
```

Comment:
```suggestion
    res$[text]truncated <- utils::getParseText(res, res$id[truncated])
```

So the left and right sides are more parallel?

Reply:
Sure, done.